### PR TITLE
[FLINK-32856][JUnit5 Migration] Migrate the testutils, throughput and throwable packages of flink-runtime module to junit5

### PR DESCRIPTION
**Pull request by 1996fanrui**
[FLINK-32856][JUnit5 Migration] Migrate the testutils, throughput and throwable packages of flink-runtime module to junit5

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/testutils/PseudoRandomValueSelectorTest.java [hunk 1]
```diff
@@ -34,25 +34,20 @@
...
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assume.assumeFalse;
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assume.assumeNoException;
 import static org.junit.Assume.assumeNotNull;
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
 
 /** Tests {@link PseudoRandomValueSelector}. */
-public class PseudoRandomValueSelectorTest extends TestLogger {
+class PseudoRandomValueSelectorTest extends TestLogger {
```
Comment (X-czh) [inline flink-runtime/src/test/java/org/apache/flink/runtime/testutils/PseudoRandomValueSelectorTest.java:43]: We should replace usage of TestLogger with TestLoggerExtension

Comment (1996fanrui) [reply flink-runtime/src/test/java/org/apache/flink/runtime/testutils/PseudoRandomValueSelectorTest.java:43]: Thanks @X-czh  for the review, I will remove the `TestLogger` later. And this PR isn't to ready, I mark it to a draft PR.

I will let you know after it's ready.

File: flink-runtime/src/test/java/org/apache/flink/runtime/testutils/TestJvmProcess.java [hunk 2]
```diff
@@ -316,18 +317,16 @@ public static void waitForMarkerFile(File file, long timeoutMillis)
...
     }
 
     public static void killProcessWithSigTerm(long pid) throws Exception {
         // send it a regular kill command (SIG_TERM)
         final Process kill = Runtime.getRuntime().exec("kill " + pid);
         kill.waitFor();
-        if (kill.exitValue() != 0) {
-            fail("failed to send SIG_TERM to process " + pid);
-        }
+        assertThat(kill.exitValue()).as("failed to send SIG_TERM to process " + pid).isZero();
```
Comment (X-czh) [inline flink-runtime/src/test/java/org/apache/flink/runtime/testutils/TestJvmProcess.java:329]: same here

**Comment by X-czh** [comment]
Thanks for your contribution! Overall LGTM with minor comments.

File: flink-runtime/src/test/java/org/apache/flink/runtime/testutils/TestJvmProcess.java [hunk 3]
```diff
@@ -316,18 +317,16 @@ public static void waitForMarkerFile(File file, long timeoutMillis)
             Thread.sleep(10);
         }
 
-        if (!exists) {
-            fail("The marker file was not found within " + timeoutMillis + " msecs");
-        }
+        assertThat(exists)
+                .as("The marker file was not found within " + timeoutMillis + " msecs")
```
[the remark this answers is not in the document]
Comment (1996fanrui) [reply flink-runtime/src/test/java/org/apache/flink/runtime/testutils/TestJvmProcess.java]: Thanks @X-czh 's review, updated!

**Comment by X-czh** [comment]
Thanks, LGTM

**Pull request merged**